included pglearn correlated load perturbations

Tamaragov opened the pull request:
updated param handler to include "correlated_scaler"
load functions to perform pglearn load perturbations (not from aggregated load profile, not time dependent)
updated default.yaml to include upper and lower limits for these load perturbations
tested data generation for 2 grids

albanpuech commented on GridDataGen/utils/load.py:
```diff
@@ -324,3 +324,64 @@ def __call__(self, net, n_scenarios, scenario_log):
+        self.l = l
+
+    def __call__(self, net, n_scenarios, scenarios_log):
+        """Generate correlated load profiles for a power grid based on aggregated load data."""
```
```suggestion
        """Generate correlated load profiles for a power grid using global and local load scaling."""
```

Tamaragov replied:
updated

albanpuech commented on scripts/config/default.yaml:
```diff
@@ -12,6 +12,8 @@ load:
   max_scaling_factor: 4.0 # Max upper bound of the global scaling factor
   step_size: 0.025 # Step size when finding the upper bound of the global scaling factor
   start_scaling_factor: 0.8 # Initial value of the global scaling factor
+  upper_limit : 1.68 # Global upper bound
+  lower_limit : 0.86 # Global lower bound 
```
what's the reason for using these limits? just curious

Tamaragov replied:
removed these limits to use find_limits function in load.py

Tamaragov commented:
updated requirements.txt to include pyYAML
removed limits from config file 
updated load.py and param_handler.py to use for find_limits function for correlated_scaler generator
